File: Python/Recursion-1/sum_digits.py

# given a non-negative int n, return the sum of its digits recursively
def sum_digits(n):
    if n == 0:
        return 0
    
    return n%10 + sum_digits(int(n/10))

print(sum_digits(126))
print(sum_digits(49))
print(sum_digits(12))

File: Java/Recursion-1/sumDigits.java

public class recursion1 {
	
	// given a non-negative int n, return the sum of its digits recursively
	public int sumDigits(int n) {
		
		if (n == 0) {
			return 0;
		}
		
		return n % 10 + sumDigits(n/10);
		
	}

	public static void main(String[] args) {

		recursion1 o = new recursion1();
		
		System.out.println(o.sumDigits(126));
		System.out.println(o.sumDigits(49));
		System.out.println(o.sumDigits(12));
		
	}

}
